### PR TITLE
feat: Allow people to opt-out on repaint boundary

### PR DESCRIPTION
**Pull request by renancaraujo**
# Description

Closes: https://github.com/flame-engine/flame/issues/2340

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/game/game_render_box.dart [hunk 1]
```diff
@@ -56,8 +63,18 @@ class GameRenderBox extends RenderBox with WidgetsBindingObserver {
...
+    if (_isRepaintBoundary == value) {
+      return;
+    }
+    _isRepaintBoundary = value;
+    markNeedsCompositingBitsUpdate();
+  }
+
   @override
-  bool get isRepaintBoundary => true;
+  bool get isRepaintBoundary => _isRepaintBoundary;
```
Comment (spydon) [inline packages/flame/lib/src/game/game_render_box.dart:77]: Maybe rename to `hasRepaintBoundary`?

Comment (renancaraujo) [reply packages/flame/lib/src/game/game_render_box.dart:77]: this is an override from RenderObject

Comment (spydon) [reply packages/flame/lib/src/game/game_render_box.dart:77]: Aah ofc 👍

**Review by spydon — APPROVED** [review]

File: packages/flame/lib/src/game/game_widget/game_widget.dart [hunk 2]
```diff
@@ -71,6 +71,9 @@ class GameWidget<T extends Game> extends StatefulWidget {
   final MouseCursor? mouseCursor;
   final List<String>? initialActiveOverlays;
 
+  /// Whether the game should assume the behavior of a [RepaintBoundary].
+  ///
+  /// Defaults to `true`.
```
Comment (spydon) [inline packages/flame/lib/src/game/game_widget/game_widget.dart:76]: Are these the docs mentioned in the commit? 😄 
We should probably add a short paragraph about it in the real docs too.

**Review by erickzanardo — APPROVED** [review]

File: packages/flame/lib/src/game/game_widget/game_widget.dart (hunk 2 above)
Comment (renancaraujo) [reply packages/flame/lib/src/game/game_widget/game_widget.dart:76]: I didn't find a proper place to add it on the MD files. Also, we don't doc most of the features of the game widget anywhere besides the dart docs.

Comment (st-pasha) [reply packages/flame/lib/src/game/game_widget/game_widget.dart:76]: This will be fixed in #2344

**Pull request merged**